### ISSUE TITLE
Extend MarkDown

### ISSUE THREAD

**Issue by pyrog** [hotosm/osm-tasking-manager2#517]
Enhance Markdown similarly to GitHub flavor.

See:
- http://en.wikipedia.org/wiki/Markdown#Extensions
- https://guides.github.com/features/mastering-markdown/

OSMTM use an "old" version of [showdown](https://github.com/showdownjs/showdown). The up to date  version support "[extensions](https://github.com/showdownjs/showdown/tree/master/src/extensions)".

A lot of GitHub flavor work actually fine, but some useful don't in OSMTM:

| Name | Markdown | Example |
| :-- | :-- | :-- |
| Strikethrough | `~~this is striked~~` | ~~this is striked~~ |
| Tables |  | this text is in a table |
| Emoji | `:bulb:` | :bulb: |
| Username | `@pyrog` | : @pyrog |
| Task Lists |  | see below |
- [x] Download JOSM Preset
- [x] Load imagery
- [ ] Load gpx file

We could also use the syntax to link issues in github to link to projects in OSMTM
 `# 515` #515 

:gb: [South Sudan Crisis, Cholera outbreak in Juba](http://tasks.hotosm.org/project/591?_LOCALE_=en)
:warning: Check in the vector data is correctly aligned on Bing imagery

:fr: [Crise au Soudan du Sud, épidémie de choléra à Juba](http://tasks.hotosm.org/project/591?_LOCALE_=fr)
:warning: Vérifier si la donnée vecteur est correctement alignée sur l'imagerie Bing

**Comment by ethan-nelson** [comment]
The sanitation currently used for displaying the project description can be configured to exclude certain styles and/or attributes.

My recent pull request #494 actually reverses the order such that the description is sanitized before markdown formatting though (there was previously an issue that broke urls with an ampersand in them, but I found a workaround). So, if these are simply extensions on the showdown, it should be easy to implement them now assuming the pull request goes through.

**Comment by pyrog** [comment]
> There are certainly ways to make sanitation without erasing everything.

There was issues that *"broke"* styles in projects descriptions in the past?

One solution is to copy some samples in the test database and/or add nose tests that check that there is no code regression.

> provide this through the T1 -- T5 tags.

This is not ![Markdown](http://upload.wikimedia.org/wikipedia/commons/thumb/4/48/Markdown-mark.svg/64px-Markdown-mark.svg.png) tags :wink: .
* *"The goal [of MD is] enabling people __to write using an easy-to-read, easy-to-write plain text format, and optionally convert it to structurally valid […] HTML__*
* *"the language was designed to be readable as-is, without looking like it's been marked up with tags or formatting instructions, unlike text formatted with a markup language"*
* Source: http://en.wikipedia.org/wiki/Markdown#History

> Any possibility with this markdown to use features such as ?

With **showdown extensions** and/or **.css** it's *"easy"* to add what you want :smile: 

But it's not clear for me what you want exactly ?
* what do you want to do ? (not how could we do that :wink:)
*  what the idea behind that:
 * emphasize important things ?
 * warn the contributor about specific tasks ?
* could you provide real examples please ?

Thanks @pierzen

**Comment by pierzen** [comment]
Yes the purpose is to emphasize , warn, show button styles for menu links to the wiki page for example.
See http://tasks.hotosm.org/project/469/edit

We did show buttons at the top, use emphasize paragraphs. A job is not static. We sometimes modify the instructions, warn contributors about an important aspect, etc.

These options were lost with the later implementation of V2, but you should see the effect copy/pasting the code in a HTML editor.

**Comment by pyrog** [comment]
Sorry Pierre, I haven't the edit privileges so I see nothing. Could you cut and past the the text inside a code section:
```markdown
Hello *world* this is a **sample** test
with [link](wikipedia.org)
```
or
```html
<p>
Hello <i>world</i> this is a <b>sample</b> test<br/>
with <a href="http://wikipedia.org">link</a>
</p>
```

<p>
Hello <i>world</i> this is a <b>sample</b> test<br/>
with <a href="http://wikipedia.org">link</a>
</p>

**Comment by pyrog** [comment]
> We did show buttons at the top, use emphasize paragraphs.

Buttons ? I will understand when I could see your 'code' :wink: 

> A job is not static.

:+1:

**Comment by pierzen** [comment]
See this html page for examples
 http://pierzen.dev.openstreetmap.org/doc/tm-html-examples.html

**Comment by pyrog** [comment]
OSMTM use in fact 2 libs for managed MD:
* frontend: [showdown](https://github.com/showdownjs/showdown) written in Javascript<br/>(Could be replaced by [marked](https://github.com/chjj/marked) a MD parser and compiler)

* backend: [markdown](https://github.com/waylan/Python-Markdown) with [extensions](https://pythonhosted.org/Markdown/extensions/index.html) written in Python<br/>(Could be replaced by [markdown2](https://github.com/trentm/python-markdown2) with [extras](https://github.com/trentm/python-markdown2/wiki/Extras))

Many solutions with (dis)advantages:
 1. Only Javascript lib
   * in the frontend only
   * in backend only (need to use nodes.js on the server)
   * in both frontend and backend
 1. Only Python lib
  * in the backend only (we need to add ajax calls for previews in the frontend)
  * in both frontend and backend
    - python to javascript compiler for the backend ([PythonJS](https://github.com/PythonJS/PythonJS ), [pyjs](http://pyjs.org/))
    - python interpreter for the frontend ([Brython](http://www.brython.info/))
 1. Keep both (as actually) but we should check that both have the same "extensions" and behaviors

For @pierzen needs, theses libs could probably accept html tags, but I prefer to use a MD syntax like the [Admonition extension](https://pythonhosted.org/Markdown/extensions/admonition.html)
